src/glang/lib/std/sys/stat.c3

```diff
@@ -0,0 +1,47 @@
+// sys/stat.h - data returned by the stat() function
+// https://pubs.opengroup.org/onlinepubs/009695399/basedefs/sys/stat.h.html
+@require_once "../wrappers.c3"
+@require_once "syscalls.c3"
+@require_once "types.c3"
+
+// This is VERY difficult to get right. fstat takes a `stat_t` buffer BUT we
+// don't actually know the layout of `stat_t` because it depends on the kernel's
+// compile flags. POSIX just guarantees that the struct contains a minimum
+// subset of fields.
+
+// https://man7.org/linux/man-pages/man3/stat.3type.html
+// The below structs are generated via the following command:
+// echo "#include <sys/stat.h>" | g++ -xc -E -
```

Comment:
I feel like the only sane/sustainable approach here is to auto-generate code like this during bootstrap/package build time.

Reply:
Yeah, I was thinking of that... The problem is libc doesn't actually need to match the syscall's memory layout... We'd need to peek into the kernel source to autogenerate.

Reply:
That's fine - I don't think getting the kernel headers is any harder than the libc headers